nano_graphrag/graphrag.py

```diff
@@ -264,20 +266,22 @@ async def ainsert(self, string_or_strings):
 
             # ---------- chunking
             inserting_chunks = {}
-            for doc_key, doc in new_docs.items():
-                chunks = {
-                    compute_mdhash_id(dp["content"], prefix="chunk-"): {
-                        **dp,
-                        "full_doc_id": doc_key,
-                    }
-                    for dp in self.chunk_func(
-                        doc["content"],
-                        overlap_token_size=self.chunk_overlap_token_size,
-                        max_token_size=self.chunk_token_size,
-                        tiktoken_model=self.tiktoken_model_name,
-                    )
-                }
-                inserting_chunks.update(chunks)
+            
+
+            
+            new_docs_list=list(new_docs.items())
+            docs=[new_doc[1]["content"] for new_doc in new_docs_list]
+            doc_keys=[new_doc[0] for new_doc in new_docs_list]
+            
+            
+            ENCODER = tiktoken.encoding_for_model("gpt-4o")
+            tokens=ENCODER.encode_batch(docs,num_threads=16)
+            chunks=self.chunk_func(tokens,overlap_token_size=self.chunk_overlap_token_size,
+                        max_token_size=self.chunk_token_size,doc_keys=doc_keys,tiktoken_model=ENCODER)
```

Comment:
Can we abstract this block into a function in `_op`? Make insert process clean and short

Reply:
I don't recommend adding another layer of abstraction here. While this unavoidable code does look a bit tedious, it doesn't significantly increase the line count. If we were to further abstract this section, we'd likely need to name this entire behavior. It almost represents a complete phase（chunking） and is hard to consider as only a single operator (_op).  , I'm still open to hearing your thoughts on this. But I might need some more detailed ideas.

Reply:
Maybe we can change the abstract of chunk_func? 
You can directly pass the `new_docs` to chunk_func and return the same things as now. So the parameters of chunk func is much more clean, so as the main process of insert .